pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -464,6 +464,12 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 	// function in order to maintain the invariant that "if the forwarding rule
 	// exists, the LB has been fully created".
 	ipAddress := ""
+	// Through this process we try to keep track of whether it is safe to
+	// release the IP that was allocated.  If the user specifically asked for
+	// an IP, we assume they are managing it themselves.  Otherwise, we will
+	// release the IP in case of early-terminating failure or upon successful
+	// creating of the LB.
+	releaseStaticIP := true
```

Comment:
Would you mind defaulting this to false? I'd prefer to have future bugs accidentally leak IPs (which we'll notice in our e2es at some point) than accidentally lose a load balancer's IP address (which a customer may notice before us)